File: 剑指offer（python）/二叉树中和为某一值的路径 .py
class Solution:
    # 返回二维列表，内部每个列表表示找到的路径
    def FindPath(self, root, expectNumber):
        # write code here
        if not root:
            return []
        result = []
        
        def FindResPath(root,path,sum):
            path.append(root)
            if sum == root.val and root.left == None and root.right == None:
                onePath=[]
                for node in path:
                    onePath.append(node.val)
                result.append(onePath)
            else:
                if root.left:
                    FindResPath(root.left,path,sum-root.val)
                if root.right:
                    FindResPath(root.right,path,sum-root.val)
            path.pop()
        
        FindResPath(root,[],expectNumber)
        return result

File: 剑指offer（c++）/二叉树中和为某一值的路径 .cpp
class Solution {
public:
    vector<vector<int> > res;
    vector<int> path;
    void find(TreeNode* root, int sum){
        if(root == NULL) return;
        path.push_back(root->val);
        if(sum == root->val && !root->right && !root->left)
            res.push_back(path);
        else{
                find(root->left,sum - root->val);
                find(root->right,sum - root->val);
        }
        path.pop_back();
            
    }
    vector<vector<int> > FindPath(TreeNode* root,int expectNumber) {
        find(root,expectNumber);
        return res;
    }
};